File: tests/test_utils.py
# -*- coding: utf-8 -*-
"""
test_utils.py - Shared test utilities for ZenAI
"""

import os
from pathlib import Path


def _default_models_dir() -> Path:
    """Resolve models directory: env var > config > project-local > fallback."""
    env = os.environ.get("ZENAI_MODEL_DIR", "")
    if env and Path(env).is_dir():
        return Path(env)
    # Try config system
    try:
        from config import MODEL_DIR
        if MODEL_DIR.is_dir():
            return MODEL_DIR
    except Exception:
        pass
    # Project-local models/
    local = Path(__file__).resolve().parent.parent / "models"
    if local.is_dir():
        return local
    return local  # return even if missing; callers check existence


def scan_models(models_dir=None):
    """Scan for available GGUF models in a directory and return Path objects."""
    if models_dir is None:
        models_path = _default_models_dir()
    else:
        models_path = Path(models_dir)
    if not models_path.exists():
        return []
    return list(models_path.glob("*.gguf"))

File: rust_output/src/test_utils.rs
/// test_utils::py - Shared test utilities for ZenAI

use anyhow::{Result, Context};
use std::path::PathBuf;

/// Resolve models directory: env var > config > project-local > fallback.
pub fn _default_models_dir() -> Result<PathBuf> {
    // Resolve models directory: env var > config > project-local > fallback.
    let mut env = std::env::var(&"ZENAI_MODEL_DIR".to_string()).unwrap_or_default().cloned().unwrap_or("".to_string());
    if (env && PathBuf::from(env).is_dir()) {
        PathBuf::from(env)
    }
    // try:
    {
        // TODO: from config import MODEL_DIR
        if MODEL_DIR.is_dir() {
            MODEL_DIR
        }
    }
    // except Exception as _e:
    let mut local = (PathBuf::from(file!()).canonicalize().unwrap_or_default().parent().unwrap_or(std::path::Path::new("")).parent().unwrap_or(std::path::Path::new("")) / "models".to_string());
    if local.is_dir() {
        local
    }
    Ok(local)
}

/// Scan for available GGUF models in a directory and return Path objects.
pub fn scan_models(models_dir: String) -> () {
    // Scan for available GGUF models in a directory and return Path objects.
    if models_dir.is_none() {
        let mut models_path = _default_models_dir();
    } else {
        let mut models_path = PathBuf::from(models_dir);
    }
    if !models_path.exists() {
        vec![]
    }
    models_path.glob("*.gguf".to_string()).into_iter().collect::<Vec<_>>()
}
